src/services/safe-wallet-provider/useSafeWalletProvider.tsx

```diff
@@ -51,13 +52,29 @@ export const _useTxFlowApi = (chainId: string, safeAddress: string): WalletSDK |
       const { title, options } = NotificationMessages.SIGNATURE_REQUEST(appInfo)
       showNotification(title, options)
 
-      return new Promise((resolve) => {
-        const unsubscribe = safeMsgSubscribe(SafeMsgEvent.SIGNATURE_PREPARED, ({ requestId, signature }) => {
+      return new Promise((resolve, reject) => {
+        let unsubscribeSignaturePrepared: (() => void) | null = null
+        let unsubscribeUserQuit: (() => void) | null = null
```

Comment:
Why is this needed? Can't you just delcare them on assignment?

Reply:
Yes, it's addressed in 31f019ca951be8f57093fdfda0d388e3a1ba0f65.